lock/lock.go

```diff
@@ -110,3 +102,92 @@ func (d *Driver) ExecuteWithMultiUseLock(ucs []config.UseLocker, timeout time.Du
 
 	return err
 }
+
+// AcquireWithTTLRefresh accepts a UseLocker, and attempts to acquire the
+// lock. When it successfully does, it then spawns a goroutine to refresh the
+// lock after a timeout, returning a stop channel. Timeout is jittered to
+// mitigate thundering herd problems.
+func (d *Driver) AcquireWithTTLRefresh(uc config.UseLocker, ttl, timeout time.Duration) (chan struct{}, error) {
+	if err := d.acquire(uc, ttl, timeout); err != nil {
+		return nil, err
+	}
+
+	stopChan := make(chan struct{})
+
+	go func() {
+		for {
+			time.Sleep(wait.Jitter(ttl/4, 0))
+			select {
+			case <-stopChan:
+				return
+			default:
+				if err := d.acquire(uc, ttl, timeout); err != nil {
+					log.Errorf("Could not acquire lock %v: %v", uc, err)
+					return
+				}
+			}
+		}
+	}()
+
+	return stopChan, nil
+}
+
+func (d *Driver) lockWait(uc config.UseLocker, timeout time.Duration, now time.Time, reason string) (bool, error) {
+	log.Warnf("Could not %s %q lock for %q", reason, uc.GetReason(), uc.GetVolume())
+	if timeout != 0 && (timeout == -1 || time.Now().Sub(now) < timeout) {
+		log.Warnf("Waiting 100ms for %q lock on %q to free", uc.GetReason(), uc.GetVolume())
+		time.Sleep(wait.Jitter(100*time.Millisecond, 0))
+		return true, nil
+	} else if time.Now().Sub(now) >= timeout {
```

Comment:
`time.Since(now)` is a [shorthand way to do this](https://golang.org/pkg/time/#Since)